x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/columns.tsx

```diff
@@ -66,15 +66,21 @@ export const getTaskStateBadge = (
 
 export const progressColumn = {
   name: i18n.translate('xpack.ml.dataframe.analyticsList.progress', {
-    defaultMessage: 'Progress',
+    defaultMessage: 'Progress per Step',
   }),
   sortable: (item: DataFrameAnalyticsListRow) => getDataFrameAnalyticsProgress(item.stats),
   truncateText: true,
   render(item: DataFrameAnalyticsListRow) {
-    const progress = getDataFrameAnalyticsProgress(item.stats);
+    const totalSteps = item.stats.progress.length;
```

Comment:
Is there any chance that any of these `item.stats.progress.length` in the chaining will be undefined?

Reply:
This should be safe, it's what we expect to get back for each step as stats.